### PR TITLE
Set environment variable `DOCS_RS` when building crates

### LINKED ISSUE #147: Detecting docs.rs from build.rs?
I have a crate that requires complicated system dependencies to build. However, the real functionality is not actually needed for docs, so I think I could fake it just for doc generation. 

Can I somehow detect docs.rs and build my crate differently for docs?

### PR DESCRIPTION
**Pull request by ldm0**
cc <https://github.com/rust-lang/docs.rs/issues/147#issuecomment-602049736>.

Fixes #147 .

### CODE REVIEW

File: tera-templates/core/about.html [hunk 1]
```diff
@@ -26,6 +26,10 @@ <h1 id="crate-title">About Docs.rs</h1>
             <a href="{{ docsrs_repo }}/issues">issues page</a>
             and file a new issue if you don't see an existing request.
             {%- endif -%}
+
+            Docs.rs builds crates with environment variable `DOCS_RS` set to
```
Comment (jyn514) [inline tera-templates/core/about.html:30]:
```suggestion
            </p>
            <p>
            Docs.rs builds crates with environment variable `DOCS_RS` set to
```

Comment (jyn514) [reply tera-templates/core/about.html:30]: This is what it looks like currently:
image

**Comment by jyn514** [comment]
Yes, this looks fine locally (except for me missing some styles which is a separate bug):

image
image
image

File: tera-templates/core/about.html [hunk 2]
```diff
@@ -28,6 +28,12 @@ <h1 id="crate-title">About Docs.rs</h1>
             {%- endif -%}
         </p>
 
+        <p>
+            Docs.rs builds crates with environment variable `DOCS_RS` set to
```
Comment (jyn514) [inline tera-templates/core/about.html:32]: One last nit:
```suggestion
            Docs.rs builds crates with environment variable <code>DOCS_RS</code> set to
```

File: tera-templates/core/about.html (hunk 1 above)
Comment (ldm0) [reply tera-templates/core/about.html]: Sorry for that, nits applied.

File: tera-templates/core/about.html (hunk 2 above)
Comment (jyn514) [reply tera-templates/core/about.html]: Please change `1` as well.

File: tera-templates/core/about.html [hunk 3]
```diff
@@ -28,6 +28,12 @@ <h1 id="crate-title">About Docs.rs</h1>
             {%- endif -%}
         </p>
 
+        <p>
+            Docs.rs builds crates with environment variable <code>DOCS_RS</code> set to
+            `1`, which enables the crate to detect docs.rs and build the
```
Comment (jyn514) [inline tera-templates/core/about.html:33]: I'll just add it myself if you don't mind :P
```suggestion
            <code>1</code>, which enables the crate to detect docs.rs and build the
```

**Comment by jyn514** [comment]
👌 

image

**Comment by jyn514** [comment]
Just deployed :)

**Pull request merged**